classifier.py

```diff
@@ -202,42 +202,72 @@ def default_target_age_for_audience(cls, audience):
         return cls.nr(None, None)
 
     @classmethod
-    def default_audience_for_target_age(cls, target_age):
-        """The default audience for a given target age.
-
-        Inverse of default_target_age_for_audience.
-        """
-        if not target_age:
-            # We were not passed a NumericRange
+    def default_audience_for_target_age(cls, nr):
+        if nr is None:
             return None
+        lower = nr.lower
+        upper = nr.upper
+        if not lower and not upper:
+            return None
+        if lower and not nr.lower_inc:
+            lower += 1
+        if upper and not nr.upper_inc:
+            upper += 1
+        if not lower:
+            if upper > 18:
+                # e.g. "up to 20 years", though that doesn't
+                # make much sense.
+                return cls.AUDIENCE_ADULT
+            elif upper > cls.YOUNG_ADULT_AGE_CUTOFF:
+                # e.g. "up to 15 years"
+                return cls.AUDIENCE_YOUNG_ADULT
+            else:
+                # e.g. "up to 14 years"
+                return cls.AUDIENCE_CHILDREN
+
+        # At this point we can assume that lower is not None.
+        if lower >= 18:
+            return cls.AUDIENCE_ADULT
+        elif lower >= cls.YOUNG_ADULT_AGE_CUTOFF:
+            return cls.AUDIENCE_YOUNG_ADULT
+        elif lower >= 12 and (not upper or upper >= cls.YOUNG_ADULT_AGE_CUTOFF):
+            # Although we treat "Young Adult" as starting at 14, many
+            # outside sources treat it as starting at 12. As such we
+            # treat "12 and up" or "12-14" as an indicator of a Young
+            # Adult audience, with a target age that overlaps what we
+            # consider a Children audience.
+            return cls.AUDIENCE_YOUNG_ADULT
+        else:
+            return cls.AUDIENCE_CHILDREN
 
-        lower = target_age.lower
-        upper = target_age.upper
+    @classmethod
```

Comment:
There were two partial implementations of this code; I refactored it and made it complete.